### ISSUE TITLE
Call to a member function getId() on null in lib/php/Report/ReportUtils.php

### ISSUE THREAD

**Issue by alpianon** [fossology/fossology#2715]
### Description

While performing a bulk spdx export on many uploads, I ran into the following error on some uploads:

```
PHP Fatal error:  Uncaught Error:  Call to a member function getId() on null in /usr/local/share/fossology/lib/php/Report/ReportUtils.php
```

#### How to reproduce

Unfortunately, I'm not allowed to provide information on the involved upload for confidentiality reasons, but I already fixed the issue, apparently (see in the comments below)

#### Screenshots

n/a

### Versions

Fossology 4.4.0 built from sources on Ubuntu 22.04

#### Job logs

```
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "PHP Fatal error:  Uncaught Error: Call to a member function getId() on null in /usr/local/share/fossology/lib/php/Report/ReportUtils.php:269"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "Stack trace:"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "#0 /usr/local/share/fossology/spdx2/agent/spdx2.php(334): Fossology\Lib\Report\ReportUtils->getFilesWithLicensesFromClearings()"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "#1 /usr/local/share/fossology/spdx2/agent/spdx2.php(228): Fossology\SpdxTwo\SpdxTwoAgent->renderPackage()"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "#2 /usr/local/share/fossology/lib/php/Agent/Agent.php(328): Fossology\SpdxTwo\SpdxTwoAgent->processUploadId()"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "#3 /usr/local/share/fossology/spdx2/agent/spdx2.php(895): Fossology\Lib\Agent\Agent->run_scheduler_event_loop()"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "#4 /usr/local/lib/fossology/fo_wrapper(65): require('/usr/local/shar...')"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "#5 {main}"
2024-04-09 23:12:57 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: "  thrown in /usr/local/share/fossology/lib/php/Report/ReportUtils.php on line 269"
2024-04-09 23:12:58 spdx2tv [0] :: JOB[68892].spdx2tv[1274310.localhost]: agent failed, code: 255
```

**Comment by GMishx** [comment]
The issue was recently fixed in #270
Please test with latest master once.

**Comment by alpianon** [comment]
I changed line 268 in lib/php/Report/ReportUtils.php and I apparently solved the problem

```diff
          $reportLicId = "$reportedLicenseId-" . md5($customLicenseText);
          $filesWithLicenses[$clearingDecision->getUploadTreeId()]
            ->addConcludedLicense($reportLicId);
          if (!array_key_exists($reportLicId, $licensesInDocument)) {
            $licenseObj = new License($concludedLicense->getId(),
              $reportedLicenseShortname, $concludedLicense->getFullName(),
              $concludedLicense->getRisk(), $customLicenseText,
              $concludedLicense->getUrl(), $concludedLicense->getDetectorType(),
              $concludedLicense->getSpdxId());
            $licensesInDocument[$reportLicId] = (new SpdxLicenseInfo())
              ->setLicenseObj($licenseObj)
              ->setCustomText(true)
              ->setListedLicense(false);
          }
+        } elseif (! is_null($concludedLicense)) {
-        } else {
          $reportLicId = $concludedLicense->getId() . "-" .
            md5($concludedLicense->getText());
          $filesWithLicenses[$clearingDecision->getUploadTreeId()]
            ->addConcludedLicense($reportLicId);
          if (!array_key_exists($reportLicId, $licensesInDocument)) {
            $licenseObj = $this->licenseDao->getLicenseById($reportedLicenseId);
            $listedLicense = !StringOperation::stringStartsWith(
              $licenseObj->getSpdxId(), LicenseRef::SPDXREF_PREFIX);
            $licensesInDocument[$reportLicId] = (new SpdxLicenseInfo())
              ->setLicenseObj($licenseObj)
              ->setCustomText(false)
              ->setListedLicense($listedLicense);
```

**Comment by alpianon** [comment]
> The issue was recently fixed in #2707 Please test with latest master once.

thanks for the instant answer! :slightly_smiling_face:

**Comment by alpianon** [comment]
@GMishx just checked by relaunching the bulk export via REST API: issue solved, thx!
